src/Context.ts

```diff
@@ -0,0 +1,19 @@
+import Knex from 'knex';
+import { PartialRequest as Request } from './Request';
+
+export interface Config {
+  authBaseUrl?: string;
+  basicAuthUser?: string;
+  logging: {
+    mode?: string;
+    color?: boolean;
+  };
+  name?: string;
+}
+
+export default interface Context {
+  connection: Knex<any, unknown[]>;
+  config: Config;
+  request: Request;
```

Comment:
i would not pass the raw request as part of the context. for the boundary of this library (i.e. the data that we'll be giving to this library in any functions that we call), we'd need the data to already be abstracted in a way that is general and useful for the library to use. the library shouldn't need to know that it's supporting a request for restify, hapi, whatever etc... all it should know is, this is how i log stuff, this is my database connection, etc...